code/game/gamemodes/objectives/_objective.dm

```diff
@@ -0,0 +1,291 @@
+		if(!is_unique_objective(possible_target,dupe_search_range))
+			continue
+		if(possible_target in blacklist)
+			continue
+
+		if(possible_target.assigned_role == "Exploration Crew")
+			if(owner_is_exploration_crew)
+				prefered_targets += possible_target
+			else
+				//Reduced chance to get people off station
+				if(prob(70) && !owner_is_shaft_miner)
+					continue
+		else if(possible_target.assigned_role == "Shaft Miner")
+			if(owner_is_shaft_miner)
+				prefered_targets += possible_target
+			else
+				//Reduced chance to get people off station
+				if(prob(70) && !owner_is_exploration_crew)
+					continue
+
+		possible_targets += possible_target
+	if(try_target_late_joiners)
+		var/list/all_possible_targets = possible_targets.Copy()
+		for(var/datum/mind/PT as() in all_possible_targets)
+			if(!PT.late_joiner)
+				possible_targets -= PT
+		if(!possible_targets.len)
+			possible_targets = all_possible_targets
+	//30% chance to go for a prefered target
+	if(prefered_targets.len > 0 && prob(30))
+		set_target(pick(prefered_targets))
+	else if(possible_targets.len > 0)
+		set_target(pick(possible_targets))
+	else
+		set_target(null)
+	update_explanation_text()
+	return target
+
+/datum/objective/proc/is_valid_target(datum/mind/possible_target)
+	if(possible_target in get_owners())
+		return FALSE
+	if(!ishuman(possible_target.current))
+		return FALSE
+	if(possible_target.current.stat == DEAD)
+		return FALSE
+	var/target_area = get_area(possible_target.current)
+	if(!HAS_TRAIT(SSstation, STATION_TRAIT_LATE_ARRIVALS) && istype(target_area, /area/shuttle/arrival))
+		return FALSE
+	return TRUE
+
+/datum/objective/proc/find_target_by_role(role, role_type=FALSE,invert=FALSE)//Option sets either to check assigned role or special role. Default to assigned., invert inverts the check, eg: "Don't choose a Ling"
+	var/list/possible_targets = list()
+	for(var/datum/mind/possible_target as() in get_crewmember_minds())
+		if(is_valid_target(possible_target))
+			var/is_role = FALSE
+			if(role_type)
+				if(possible_target.special_role == role)
+					is_role = TRUE
+			else
+				if(possible_target.assigned_role == role)
+					is_role = TRUE
+			if(is_role && !invert || !is_role && invert)
+				possible_targets += possible_target
+	if(length(possible_targets))
+		set_target(pick(possible_targets))
+	else
+		set_target(null)
+	update_explanation_text()
+	return target
+
+/datum/objective/proc/update_explanation_text()
+	if(team_explanation_text && LAZYLEN(get_owners()) > 1)
+		explanation_text = team_explanation_text
+
+/datum/objective/proc/generate_stash(list/special_equipment)
+	var/datum/mind/receiver = pick(get_owners())
+	var/obj/item/storage/backpack/satchel/flat/empty/secret_bag = receiver.antag_stash
+	//Find and generate the stash
+	if(!secret_bag)
+		secret_bag = new()
+		// Hard-del handling is done by the component
+		receiver.antag_stash = secret_bag
+		var/atom_text = ""
+		switch (pick_weight(list("airlock" = 3)))
+			if("airlock")
```

Comment:
add potted plants and toilets in public areas like dorms and hallway

Reply:
They are publicly accessible, out of scope